scientific-computing-with-python/learn-string-manipulation-by-building-a-cipher/vigenere.py

```diff
@@ -0,0 +1,37 @@
+from string import ascii_lowercase as letters
+
+def vigenere(message, key, direction):
+    spaced_key = ''
+    key_index = 0
+    final_message = ''
+    for char in message.lower():        
+        if char == ' ':            
+            # Keep the same spacing in message and key
+            final_message += ' '            
+            spaced_key += ' '

```

Comment:
To be honest my intention was to keep things as simple as possible and don't consider special characters, since this is intended to be the first project (unless things change soon). Anyway this is still simple and straightforward so I guess it can be one of the final tweaks.